src/modules/flexible/utils.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  TripPattern,
+  TripPatternWithDetails,
+} from '@atb/page-modules/assistant/server/journey-planner/validators';
+import {
+  BookingArrangement,
+  BookingStatus,
+  TripPatternBookingStatus,
+} from './types';
+import {
+  dateWithReplacedTime,
+  iso8601DurationToSeconds,
+  secondsBetween,
+} from '@atb/utils/date';
+
+export const isLineFlexibleTransport = (
+  line?:
+    | TripPatternWithDetails['legs'][0]['line']
+    | TripPattern['legs'][0]['line'],
+) => !!line?.flexibleLineType;
+
+export const getBookingStatus = (
+  bookingArrangements: BookingArrangement | null,
+  aimedStartTime: string,
+  flex_booking_number_of_days_available?: number,
+): BookingStatus => {
+  if (!bookingArrangements) return 'none';
+
+  const secondsToDeadline = getSecondsRemainingToBookingDeadline(
+    bookingArrangements,
+    aimedStartTime,
+    Date.now(),
```

Comment:
To account for time zone differences:
```suggestion
    setTimezone(Date.now()),
```

Reply:
Entur returns UTC and `Date.now()` is UTC so this should be correct 🤔 I'll do some testing